src/features/search/use_cases/search_use_case.py

```diff
@@ -15,30 +15,23 @@ class SearchRequest(DataSourceList):
     dotted_attribute_path: str
 
 
-def get_search_result(output_dict: dict, data_source_id: str, document_service: DocumentService, req: SearchRequest):
-    results = document_service.search(
-        data_source_id=data_source_id,
-        search_data=deepcopy(req.data),
-        dotted_attribute_path=req.dotted_attribute_path,
-    )
-    return output_dict.update(results)
-
-
-def search_use_case(user: User, request: SearchRequest, repository_provider=get_data_source):
+def search_use_case(user: User, request: SearchRequest, repository_provider=get_data_source) -> dict:
     document_service: DocumentService = DocumentService(repository_provider=repository_provider, user=user)
-    all_data_sources = DataSourceRepository(user).list()
+    all_data_source_ids = [ds["id"] for ds in DataSourceRepository(user).list()]
+    search_data_sources = all_data_source_ids
 
-    search_results: dict = {}
+    if request.data_sources:
+        # If user has specified any data sources, check that they exist, then select only through them for search.
+        if invalid_search_data_sources := set(request.data_sources) - set(all_data_source_ids):
+            raise BadRequestException(f"Data source {invalid_search_data_sources.pop()} not found")
```

Comment:
If there are more than one invalid data source, it might be better to print the ids of all of the invalid ones, not just the first one.